Python/import.c

```diff
@@ -885,7 +869,25 @@ PyObject *
 PyImport_AddModuleObject(PyObject *name)
 {
     PyThreadState *tstate = _PyThreadState_GET();
-    return import_add_module(tstate, name);
+    PyObject *mod = import_add_module(tstate, name);
+    if (mod) {
+        if (Py_REFCNT(mod) == 1) {
+            /* This check does not preven an undefined behavior in the
+             * following code, because the module can have references
+             * to itself. */
```

Comment:
```suggestion
            /* This check does not prevent an undefined behavior in the
             * following code, because the module can have references
             * to itself. */
```